### PR TITLE
Clean up Users class; store Users in datastore on login; add React community page.

### PR DESCRIPTION
**Pull request by tylerhou**

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -78,22 +82,25 @@ public void storeMessage(Message message) {
   /**
    * Get List of messages posted by a specific user.
    *
-   * @return a list of messages posted by the user, or empty list if user has never posted a
+   * @return a list of messages posted by the user, or empty list if user has
+   *     never posted a
```
Comment (jajanet) [inline src/main/java/com/google/codeu/data/Datastore.java]: formatting here is a bit weird i think?

File: src/main/java/com/google/codeu/data/User.java [hunk 2]
```diff
@@ -1,21 +1,77 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
 public class User {
 
-    private String email;
-    private String aboutMe;
+  private String email;
+  private String aboutMe;
+  private transient String id;
```
Comment (jajanet) [inline src/main/java/com/google/codeu/data/User.java]: i looked up what transient was in java, but i'm curious about the use case here and whether we need to add it to other files?

Comment (tylerhou) [reply src/main/java/com/google/codeu/data/User.java]: Transient means that Gson won't serialize it when sending it down to the client. I don't know whether exposing user IDs is sensitive or not, so I'm choosing not to send it down. I don't think there are other places where we need to add `transient`.

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (tylerhou) [reply src/main/java/com/google/codeu/data/Datastore.java]: My formatter borked, I've updated it. (It was breaking the comment at 80 chars)

**Comment by tylerhou** [comment]
> likely should've separate into different PRs since they go under different branches and serve different functions 

@jajanet These are all the same feature -- I had to create the community page. To create the community page I need a list of users; we don't have messages anymore (or we will be getting rid of messages) so to find the list of users I need to store them in Datastore; at the same time I might as well also clean up the Users class because it makes my code easier to write.

You're right that I maybe should have split them up into different PRs, but in this case I thought it was fine for a single PR since they were 1) a chain of related features and 2) a lot of it was copying code around, so it's easier to review. In any case, all of the features here would eventually have to make their way into master.

I think you're trying to say that I enforced a policy of 1 pr = 1 feature wrt. PR #26. In that case styles and the bug fix were unrelated, and neither blocked the other -- as in, to get styles in you don't need to fix the bug, and to fix the bug you don't need to add styles. In that case I asked you to split the PR in two because we want to get the bugfix into master, but pulling in Bootstrap and jquery is not your individual decision and rather a team decision.

> since that's the protocol that was being enforced the past few weeks

I don't think this protocol has been enforced; 9 days ago we merged your PR (#34) which was a bunch of mismashed features.
  
Again your criticism is right -- in an ideal world, I should have split these into a chain of PRs. But it's not a huge deal. To recap, I asked you to split #26 because one of the features in that PR *should not* be merged into master (or needed more discussion).

**Pull request merged**